src/api/app/models/bs_request_permission_check.rb

```diff
@@ -193,25 +193,25 @@ def set_permissions_for_action(action)
     end
 
     # general source write permission check (for revoke)
-    if (@source_package and User.current.can_modify_package?(@source_package, true)) or
-        (not @source_package and @source_project and User.current.can_modify_project?(@source_project, true))
+    if (@source_package && User.current.can_modify_package?(@source_package, true)) ||
+        (!@source_package && @source_project && User.current.can_modify_project?(@source_project, true))
       @write_permission_in_source = true
     end
 
     # general write permission check on the target on accept
     @write_permission_in_this_action = false
     # meta data change shall also be allowed after freezing a project using force:
-    ignoreLock = opts[:force] and [:set_bugowner].include? action.action_type
+    (ignoreLock = opts[:force]) && [:set_bugowner].include?(action.action_type)
```

Comment:
Do you want that I correct this in this PR? I would prefer doing it in a separate one.